File: admin-portal/src/services/order.service.ts
import api from './api';
import { Order, OrderTrackingSnapshot } from '../types';

export const orderService = {
  async getAll(): Promise<Order[]> {
    const response = await api.get<Order[]>('/orders');
    return response.data;
  },

  async getByRestaurant(restaurantId: string): Promise<Order[]> {
    const response = await api.get<Order[]>(`/orders/restaurant/${restaurantId}`);
    return response.data;
  },

  async getById(id: string): Promise<Order> {
    const response = await api.get<Order>(`/orders/${id}`);
    return response.data;
  },

  async updateStatus(id: string, status: Order['status']): Promise<Order> {
    const response = await api.patch<Order>(`/orders/${id}/status`, { status });
    return response.data;
  },

  async assignRider(id: string, riderId: string): Promise<{ order: Order }> {
    const response = await api.patch<{ order: Order }>(`/orders/${id}/assign-rider`, {
      riderId,
    });
    return response.data;
  },

  async getTracking(id: string): Promise<OrderTrackingSnapshot> {
    const response = await api.get<OrderTrackingSnapshot>(`/orders/${id}/tracking`);
    return response.data;
  },
};

File: rider/lib/services/order_service.dart
import '../core/api_client.dart';
import '../models/order.dart';

class OrderService {
  final _dio = ApiClient.instance.dio;

  /// Orders currently assigned to this rider (any non-final status).
  Future<List<DeliveryOrder>> myActiveOrders(String riderId) async {
    final resp = await _dio.get('/orders', queryParameters: {'driverId': riderId});
    final all = (resp.data as List<dynamic>)
        .map((e) => DeliveryOrder.fromJson(e as Map<String, dynamic>))
        .toList();
    return all.where((o) => o.status != 'delivered' && o.status != 'cancelled').toList();
  }

  Future<List<DeliveryOrder>> myDeliveredOrders(String riderId) async {
    final resp = await _dio.get('/orders', queryParameters: {
      'driverId': riderId,
      'status': 'delivered',
    });
    return (resp.data as List<dynamic>)
        .map((e) => DeliveryOrder.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  Future<DeliveryOrder> updateStatus(String orderId, String status) async {
    final resp = await _dio.patch('/orders/$orderId/status', data: {'status': status});
    return DeliveryOrder.fromJson(resp.data as Map<String, dynamic>);
  }

  Future<DeliveryOrder> getById(String orderId) async {
    final resp = await _dio.get('/orders/$orderId');
    return DeliveryOrder.fromJson(resp.data as Map<String, dynamic>);
  }
}
